### ISSUE TITLE
Switching from one route to another raises psEquals error

### ISSUE THREAD

**Issue by tcastelli** [okgrow/meteor-persistent-session#41]
Getting this error when moving between routes with 0.4.0. Didn't happen before(switched back to 0.3) I've rechecked my routes and all seem correct with their this.next();

```javascript
"Exception in callback of async function: _psEquals/<@http://localhost:3000/packages/u2622_persistent-session.js?17356c8a89f38d104bb8abb5237ce9ae72c3c932:330:13
Tracker.Computation.prototype.invalidate/<@http://localhost:3000/packages/tracker.js?6d0890939291d9780f7e2607ee3af3e7f98a3d9c:296:9
Tracker.nonreactive@http://localhost:3000/packages/tracker.js?6d0890939291d9780f7e2607ee3af3e7f98a3d9c:593:12
Tracker.Computation.prototype.invalidate@http://localhost:3000/packages/tracker.js?6d0890939291d9780f7e2607ee3af3e7f98a3d9c:295:7
Tracker.Dependency.prototype.changed@http://localhost:3000/packages/tracker.js?6d0890939291d9780f7e2607ee3af3e7f98a3d9c:421:5
.set/changed@http://localhost:3000/packages/reactive-dict.js?1cb61826ceb23464171f17c2191c7e04a9b91508:100:12
.set@http://localhost:3000/packages/reactive-dict.js?1cb61826ceb23464171f17c2191c7e04a9b91508:103:5
PersistentSession.prototype.old_set@http://localhost:3000/packages/u2622_persistent-session.js?17356c8a89f38d104bb8abb5237ce9ae72c3c932:236:10
_psSet@http://localhost:3000/packages/u2622_persistent-session.js?17356c8a89f38d104bb8abb5237ce9ae72c3c932:256:3
_psSetTemp@http://localhost:3000/packages/u2622_persistent-session.js?17356c8a89f38d104bb8abb5237ce9ae72c3c932:347:3
.onRun@http://localhost:3000/client/routers/home.js?697274f918ce873ac41a1f0d1d6131f657dd7f89:31:13
callEventHandlerOrNext@http://localhost:3000/packages/reywood_iron-router-ga.js?347c8d22ba79156c007cf33e82c8460c7b704e52:216:16
attachPageViewTracking/options.onRun@http://localhost:3000/packages/reywood_iron-router-ga.js?347c8d22ba79156c007cf33e82c8460c7b704e52:155:16
callEventHandlerOrNext@http://localhost:3000/packages/reywood_iron-router-ga.js?347c8d22ba79156c007cf33e82c8460c7b704e52:216:16
attachPageSetter/options.onRun@http://localhost:3000/packages/reywood_iron-router-ga.js?347c8d22ba79156c007cf33e82c8460c7b704e52:165:16
boundNext@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:424:16
Meteor.bindEnvironment/<@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:17
scrollToTop@http://localhost:3000/packages/okgrow_iron-router-autoscroll.js?44dcb44bdca8b44627d4ad3679b912c92257b6f3:61:3
Router.prototype.addHook/hookWithOptions/<@http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:1209:14
.withValue@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:955:17
Router.prototype.addHook/hookWithOptions@http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:1208:1
boundNext@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:424:16
Meteor.bindEnvironment/<@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:17
dispatch@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:448:3
onRun@http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:503:11
boundNext@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:424:16
Meteor.bindEnvironment/<@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:17
dispatch@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:448:3
RouteController.prototype._runRoute@http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:544:3
Route.prototype.dispatch@http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:854:10
Route/route@http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:711:5
boundNext@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:424:16
Meteor.bindEnvironment/<@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:17
boundNext@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:371:1
Meteor.bindEnvironment/<@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:17
boundNext@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:371:1
Meteor.bindEnvironment/<@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:17
boundNext@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:371:1
Meteor.bindEnvironment/<@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:17
boundNext@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:371:1
Meteor.bindEnvironment/<@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:17
boundNext@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:371:1
Meteor.bindEnvironment/<@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:17
boundNext@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:371:1
Meteor.bindEnvironment/<@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:17
dispatch@http://localhost:3000/packages/iron_middleware-stack.js?56e0c02a758f48c0a704ca978a8fb6d7bd9ebbab:448:3
RouteController.prototype.dispatch/</<@http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:391:7
Tracker.Computation.prototype._compute@http://localhost:3000/packages/tracker.js?6d0890939291d9780f7e2607ee3af3e7f98a3d9c:327:5
Tracker.Computation@http://localhost:3000/packages/tracker.js?6d0890939291d9780f7e2607ee3af3e7f98a3d9c:243:5
Tracker.autorun@http://localhost:3000/packages/tracker.js?6d0890939291d9780f7e2607ee3af3e7f98a3d9c:566:11
RouteController.prototype.dispatch/<@http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:389:5
Tracker.nonreactive@http://localhost:3000/packages/tracker.js?6d0890939291d9780f7e2607ee3af3e7f98a3d9c:593:12
RouteController.prototype.dispatch@http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:388:3
Router.prototype.dispatch@http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:1698:3
onLocationChange@http://localhost:3000/packages/iron_router.js?7b16fe70bccff9182aca02afb2ccd7708ac57c64:1782:20
Tracker.Computation.prototype._compute@http://localhost:3000/packages/tracker.js?6d0890939291d9780f7e2607ee3af3e7f98a3d9c:327:5
Tracker.Computation.prototype._recompute@http://localhost:3000/packages/tracker.js?6d0890939291d9780f7e2607ee3af3e7f98a3d9c:346:9
Tracker._runFlush@http://localhost:3000/packages/tracker.js?6d0890939291d9780f7e2607ee3af3e7f98a3d9c:485:9
onGlobalMessage@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:372:11
" meteor.js:888:11
"Exception from Tracker recompute function:" meteor.js:888:11
"TypeError: this._dict is undefined" meteor.js:888:11
"_psEquals/<@http://localhost:3000/packages/u2622_persistent-session.js?17356c8a89f38d104bb8abb5237ce9ae72c3c932:330:13" meteor.js:888:11
"Exception from Tracker recompute function:" meteor.js:888:11
"TypeError: this._dict is undefined" meteor.js:888:11
"_psEquals/<@http://localhost:3000/packages/u2622_persistent-session.js?17356c8a89f38d104bb8abb5237ce9ae72c3c932:330:13" meteor.js:888:11
"Route dispatch never rendered. Did you forget to call this.next() in an onBeforeAction?"
```

**Comment by tcastelli** [comment]
whatever it was, seems to be fixed on 0.4.1

**Comment by tcastelli** [comment]
I just got the same error while doing a logout (seems to come from useraccounts iron-router package)
```javascript
Exception in delivering result of invoking 'logout': _psClear/<@http://localhost:3000/packages/u2622_persistent-session.js?da5fb5879831cbca0be984fd856aa67f5efe127f:438:5
_.forEach@http://localhost:3000/packages/underscore.js?0a80a8623e1b40b5df5a05582f288ddd586eaa18:164:13
_psClear@http://localhost:3000/packages/u2622_persistent-session.js?da5fb5879831cbca0be984fd856aa67f5efe127f:434:3
logoutFunc@http://localhost:3000/both/begin/accounts.js?c099b065b187317afb0831587164490edadc7221:34:5
AccountsTemplates.logout/<@http://localhost:3000/packages/useraccounts_iron-routing.js?b40e5b12176b0cb51e7f2320bdce8b9e95406f33:439:7
Meteor.logout/<@http://localhost:3000/packages/accounts-base.js?7c29db5c21a76726509bb6bb2a68a2b4b1ecf657:674:19
Meteor.bindEnvironment/<@http://localhost:3000/packages/meteor.js?43b7958c1598803e94014f27f5f622b0bddc0aaf:983:17
._maybeInvokeCallback@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:3860:7
.dataVisible@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:3889:5
._runAfterUpdateCallbacks/<@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:4718:7
_.forEach@http://localhost:3000/packages/underscore.js?0a80a8623e1b40b5df5a05582f288ddd586eaa18:156:7
._runAfterUpdateCallbacks@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:4717:5
._livedata_data@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:4707:5
Meteor.connection._livedata_data@http://localhost:3000/packages/meteorhacks_fast-render.js?94fcf6cfebcd08497b06448d0ec5f2e3cb7853d8:397:10
DDPProvider._livedata_data/<@http://localhost:3000/packages/kadira_debug.js?7b37bca85e91dfc6de6d549bce3a6a5a89bf42f3:616:12
Connection/onMessage@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:3721:7
._launchConnection/self.socket.onmessage/<@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:2717:11
_.forEach@http://localhost:3000/packages/underscore.js?0a80a8623e1b40b5df5a05582f288ddd586eaa18:156:7
._launchConnection/self.socket.onmessage@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:2716:9
REventTarget.prototype.dispatchEvent@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:156:9
SockJS.prototype._dispatchMessage@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:1141:5
SockJS.prototype._didMessage@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:1199:13
SockJS.websocket/that.ws.onmessage@http://localhost:3000/packages/ddp.js?d1840d3ba04c65ffade261f362e26699b7509706:1346:9
```

**Comment by deanius** [comment]
Can you include more information, perhaps the code of the repo that exhibits this issue ?

**Comment by tcastelli** [comment]
I think i was doing a double logout so the package was trying to remove non existing session variables(auth). So it was connected with #42

**Comment by rgould** [comment]
@tcastelli We've just published v0.4.2, which fixes #42, and so should fix the case with the double logout. If you experience an issue again go ahead and re open this issue.
